app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -1184,6 +1175,27 @@ public function import($fields, $data, $owner = null, $list = null, $tags = null
         $lead ??= $this->checkForDuplicateContact($fieldData);
         $merged = (bool) $lead->getId();
 
+        // Extract company data and import separately
+        // Modifies the data array
+        $company                       = null;
+        [$companyFields, $companyData] = $this->companyModel->extractCompanyDataFromImport($fields, $data);
+
+        if ((true === $skipIfExists && empty($lead->getCompany())) || false === (bool) $skipIfExists) {
+            if (!empty($companyData)) {
+                $company = $this->companyModel->importCompany(
+                    array_flip($companyFields),
+                    $companyData,
+                    null,
+                    true,
+                    $skipIfExists
+                );
+            }
+        }
```

Comment:
I read this that if the imported contact has a company then the company fields shouldn't be imported at all. Is that how it should behave? Shouldn't it create the company but perhaps do not link it with the contact?

Reply:
I'm not sure why, but this is about contact imports. Part of the contact import process also includes importing related company data linked to the contacts. If the data isn't related to a contact, the user cannot create a company through the contact import feature. 

Companies have their own separate import process for companies. 

It seems legit to me

Reply:
Alright. I wasn't sure this was intentional. I'm not sure which way is more logic myself. Let's say this is the right way to do it.